src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_ProfileStack/Profile_PaymentOptionsScreen.tsx

```diff
@@ -90,11 +87,19 @@ export const Profile_PaymentOptionsScreen = () => {
       addPaymentMethodCallbackHandler,
     );
     return () => eventSubscription.remove();
-  }, [appStatus, authorizeRecurringPayment, cancelRecurringPayment]);
+  }, [authorizeRecurringPayment, cancelRecurringPayment]);
 
   const onAddRecurringPayment = async () => {
     const redirectUrl = `${APP_SCHEME}://profile`;
-    await addPaymentMethod(redirectUrl);
+    await addPaymentMethod(redirectUrl).then(async (response) => {
+      await InAppBrowser.open(
+        response.data.terminal_url,
+        // Param showInRecents is needed so the InAppBrowser doesn't get closed when the app goes to background
+        // hence user is again navigated back to browser after finishing the Nets flow,
+        // and then can complete the authentication process successfully
+        {showInRecents: true},
+      );
+    });
```

Comment:
This mix of `await` and `then`, with a nested `await` caught my eye. It isn't doing anything really wrong, but looks off 😅 What about just 
```ts
const response = await addPaymentMethod(redirectUrl);
await InAppBrowser.open(response.data.terminal_url, ...);
```

Reply:
Now that you mentioned it, it does look off... refactored